src/strategies/bmc/index.ts

```diff
@@ -0,0 +1,75 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { Multicaller } from '../../utils';
+
+export const author = 'justinrwong';
+export const version = '0.0.1';
+
+const ogAbi = [
+  'function walletMiners(address account) external view returns (uint256[])'
+];
+
+const ultraAbi = [
+  'function tokensOfOwner(address account) external view returns (uint256[])'
+];
+
+const stakingAbi = [
+  'function getStakedOGs(address account) external view returns (uint256[])',
+  'function getStakedULTRAs(address account) external view returns (uint256[])'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  // Setup multicall for the OG/Ultra/Staking Contracts
+  const ogToken = new Multicaller(network, provider, ogAbi, { blockTag });
+  const ultraToken = new Multicaller(network, provider, ultraAbi, { blockTag });
+  const stakingOGToken = new Multicaller(network, provider, stakingAbi, {
+    blockTag
+  });
+  const stakingUltraToken = new Multicaller(network, provider, stakingAbi, {
+    blockTag
+  });
+  addresses.forEach((address) => {
+    ogToken.call(address, options.og, 'walletMiners', [address]);
+    ultraToken.call(address, options.ultra, 'tokensOfOwner', [address]);
+    stakingOGToken.call(address, options.staking, 'getStakedOGs', [address]);
+    stakingUltraToken.call(address, options.staking, 'getStakedULTRAs', [
+      address
+    ]);
+  });
+
+  // when no assets are held, the contract throws an error, which means it's empty.
+  const [ogResponse, ultraResponse, stakedOgResponse, stakedUltraResponse]: [
+    Record<string, BigNumber[]>,
+    Record<string, BigNumber[]>,
+    Record<string, BigNumber[]>,
+    Record<string, BigNumber[]>
+  ] = await Promise.all([
+    ogToken.execute().catch((error) => []),
+    ultraToken.execute().catch((error) => []),
+    stakingOGToken.execute().catch((error) => []),
+    stakingUltraToken.execute().catch((error) => [])
```

Comment:
Hey @JustinRWong Don't catch the errors, in case of an error, users may get `0` voting power, which is not a correct voting power (we handle these errors in a top level, so should be fine :)

Reply:
What do you mean by don't catch the error? the contract itself throws the error in the call which makes the tests fail. Do you mean contracts calls that return errors are already handled by Multicall?

Reply:
Answered in discord, make sure it doesn't return any errors, remove catch part here

Reply:
Following up, since the contract themselves have a `require` statement, a [CALL_EXCEPTION](https://docs.ethers.io/v5/troubleshooting/errors/#help-CALL_EXCEPTION) error can happen, making the catch statement necessary for the tests to pass.

to resolve, this now catches only the CALL_EXCEPTION error code and would otherwise throw the error